test/OpenTelemetry.Tests/Impl/Trace/SpanTest.cs

```diff
@@ -164,7 +164,7 @@ public void StartSpanFrom_Recorded_ParentSpan_Kind_Links()
             var linkContext = new SpanContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.Recorded);
 
             var startTimestamp = PreciseTimestamp.GetUtcNow();
-            var span = (Span)tracer.StartSpan(SpanName, parentSpan, SpanKind.Server, default, new [] { new Link(linkContext) });
+            var span = (Span)tracer.StartSpan(SpanName, parentSpan, SpanKind.Server, default, () => new [] { new Link(linkContext) });
```

Comment:
I think we should have tests for links passed as list AND tests for lambda. I might have missed something, but it seems there are no tests for a simple list of links anymore, is that right?

Reply:
Yes it's missing tests, I'll take care of it.
The issue here is that tests are doing a lot which means a lot of duplication if I want to cover every user-case covered with `IEnumerable` also with `Func<>`.

Reply:
On the first pass I had removed all `IEnumerable` overloads to `Func<>` and in the processed had changed all tests to use `Func<>` instead.
I've added tests for both now.